### ISSUE TITLE
Multi-targeting: net40, net461, netstandard2.0

### ISSUE THREAD

**Issue by ksakz** [ClosedXML/ClosedXML#101]
is it meant to be?

**Comment by igitur** [comment]
The OpenXML library has to support NET Core first before we can attempt it. ClosedXML is just a wrapper around OpenXML

**Comment by igitur** [comment]
Depends on https://github.com/OfficeDev/Open-XML-SDK/issues/111

**Comment by rogersachan** [comment]
@igitur I thought supporting .NET Standard 1.0 makes it easier to support .NET Core.

**Comment by Pyropace** [comment]
@igitur it looks like 111 is closed, perhabs we can start looking into this 

this was merged in https://github.com/OfficeDev/Open-XML-SDK/pull/121

I will assign this to my self and see if i can get it compiling

**Comment by Pyropace** [comment]
Well things are looking alot better than last time i tried this. 

1- OpenXML now  is dot net standard compatible ( however an official package hasnot been released to nuget  yet,(version 3.0.0)  tried the latest version from their source  and built locally )
2- We still need corefx team to finish their work with System.Drawing library, (https://github.com/dotnet/corefx/issues/2315), however using (https://github.com/zkweb-framework/ZKWeb.System.Drawing) can get you to compile 
3- DatTables, datasets are not supported in dotnet standard, so we would have to change all the codes that use DataTable ( not a bad idea ) 
4- we use OADate alot in our code, that also has been dropped as it is windows specific, we would have to replace codes related to that as well. 

I got a version to compile, but it fails a lot of tests, I might create a branch and push what i have so far, and once OpenXml 3.0.0 is released, we can get it to work 

but we would have to make alot of code change

**Comment by MKI-Miro** [comment]
Hi, anything new ?

**Comment by moattarwork** [comment]
+1

**Comment by danilobreda** [comment]
With the new .net standard 2.0. When will it be ported to .net standard?

**Comment by christophwille** [comment]
I am also going through various projects that are candidates for Core 2.0 porting atm (OSS as well as enterprise) and ClosedXML is one that I really like & rely on heavily. As this issue is marked as Parked atm, any plans to make this happen?

**Comment by igitur** [comment]
You can follow the progress at #315 . Even better, you can help contribute.

**Comment by igitur** [comment]
Dependent on https://github.com/OfficeDev/Open-XML-SDK/issues/226

**Comment by igitur** [comment]
Preliminary .NET Standard 2.0 build can be found at https://ci.appveyor.com/project/Pyropace/closedxml/build/artifacts

The dependencies as per https://github.com/igitur/ClosedXML/blob/netstandard2/ClosedXML/ClosedXML.csproj should be included if you want to test this.

**Comment by igitur** [comment]
Sorry, my bad, it doesn't work yet.

**Comment by igitur** [comment]
Dependent on https://github.com/OfficeDev/Open-XML-SDK/issues/235

**Comment by kenleese** [comment]
As a more precise dependency you may wish to watch https://github.com/dotnet/corefx/issues/23795

**Comment by igitur** [comment]
Preliminary .NET Standard 2.0 build can be found at https://ci.appveyor.com/project/Pyropace/closedxml/build/0.8.524/artifacts

The dependencies as per https://github.com/igitur/ClosedXML/blob/netstandard2/ClosedXML/ClosedXML.csproj should be included if you want to test this.

**Comment by igitur** [comment]
@ksakz @rogersachan @danilobreda @christophwille 
I need lots of user feedback before I can merge this.

**Comment by lemoncatyang** [comment]
Hi, I have a question, how can I use the netstandard2.0 version in my project now? Really like ClosedXML, Thank you !

**Comment by igitur** [comment]
@ych-szh You have 2 options: Build the branch in #315 from source or use the build from the continuous integration as commented above in https://github.com/ClosedXML/ClosedXML/issues/101#issuecomment-327516740

**Comment by dumonjorne** [comment]
Any ETA when it will get released? (don't want to rush anyone, just nice to know)

**Comment by igitur** [comment]
@dumonjorne Before I can even think of releasing, I need user feedback. Been begging for it (see above). No response yet. This project needs more collaboration.

**Comment by kenleese** [comment]
@igitur The netstandard2 branch (built from source) passes all tests in my ASP.NET Core 2 server application.

**Comment by igitur** [comment]
Thanks @kenleese - do you mean the tests from our own, separate application that relies on ClosedXML? If so, that's good news, as chances are good your tests might cover some things that aren't covered in ClosedXML's own tests.

**Comment by kenleese** [comment]
@igitur, yes a distinct server application.
It's possible that greater test activity might be stimulated by publishing a NuGet pre-release package, even if identified by "-alpha1".
Meanwhile, thank you for continuing to merge changes into the netstandard2 branch (and, more generally, thank you for the ongoing improvements to ClosedXml).

**Comment by igitur** [comment]
Updated .NET Standard 2.0 build can be found at https://ci.appveyor.com/project/Pyropace/closedxml/build/0.8.635/artifacts

Includes a possible fix for reducing memory usage.

The dependencies as per https://github.com/igitur/ClosedXML/blob/netstandard2/ClosedXML/ClosedXML.csproj should be included if you want to test this.

**Comment by nefcanto** [comment]
This is one of the best libraries I've ever worked with in my entire 11 years of working experience. A well-defined API, very light, efficient, and truly developer-friendly. It's a good news that .NET Core version is coming out. Thank you so much guys.

**Comment by paulirwin** [comment]
Trying to test this out and it does not work for me. Getting error:

```
FileLoadException: Could not load file or assembly 'FastMember.Signed, Version=1.0.0.9, Culture=neutral, PublicKeyToken=9e8f22703bef9a29' or one of its dependencies.
```

I've tried installing FastMember.Signed NuGet version 1.1.0 as referenced in the CSPROJ you linked to, as well as 1.2.0 and no luck due to strong naming. There is no 1.0.0.9 on NuGet.

**Comment by igitur** [comment]
Try the unsigned version of FastMember

**Comment by paulirwin** [comment]
For some reason it started working after changing the target to .NET Core 2.0. It works still when I change it back to 4.7.1. Disregard my earlier comment.

The library is working well on both .NET Framework 4.7.1 and .NET Core 2.0 for me now. Granted, my use case is very simple (single sheet, single table, 2 columns) but it works.

**Comment by vpenades** [comment]
Microsoft has recently released a net standard version of System.Drawing, called System.Drawing.Common.

**Comment by igitur** [comment]
@vpenades Thanks, I also recently saw that. I'll definitely try to use that library.

**Comment by vpenades** [comment]
@igitur Additionally, you can switch to SixLabors.ImageSharp, which is a pure .Net implementation

**Comment by igitur** [comment]
The `KnownColors` enum is not found in `System.Drawing.Common`. I'm considering removing it totally from ClosedXML or maybe just leave it in there for .NET Framework only. Comments?

**Comment by vbjay** [comment]
Fire off an issue requesting it.

**Comment by igitur** [comment]
It's already been thoroughly discussed at
https://github.com/dotnet/standard/pull/290 and the linked issues.

**Comment by vpenades** [comment]
[Here's some interesting reading about keeping support for System.Drawing](http://photosauce.net/blog/post/5-reasons-you-should-stop-using-systemdrawing-from-aspnet)

**Comment by igitur** [comment]
I've released an alpha build of the `netstandard2.0` package on MyGet: https://www.myget.org/feed/closedxml/package/nuget/ClosedXML

Feedback wanted before this is going into beta.

**Comment by kenleese** [comment]
@igitur Windows Server / IIS

**Comment by igitur** [comment]
`0.92.0-beta1` has been released that now includes a .NET Standard 2.0 build. https://www.nuget.org/packages/ClosedXML/0.92.0-beta1

**Comment by igitur** [comment]
Fixed in #315
